metagraph/examples/ExternalProject/CMakeLists.txt

```diff
@@ -0,0 +1,115 @@
+cmake_minimum_required(VERSION 3.16)
+
+project(ExternalProject_Add_example)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Set MetaGraph version
+set(METAGRAPH_GIT_TAG "master" CACHE STRING "Git tag/branch/commit for MetaGraph")
+
+# Set alphabet (DNA or Protein)
+if(NOT DEFINED CMAKE_DBG_ALPHABET)
+    set(CMAKE_DBG_ALPHABET "DNA" CACHE STRING "Alphabet for MetaGraph (DNA or Protein)")
+endif()
+
+# Include ExternalProject module
+include(ExternalProject)
+
+# Forward optional build flags to MetaGraph if they're set
+set(METAGRAPH_CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+    -DCMAKE_DBG_ALPHABET=${CMAKE_DBG_ALPHABET}
+    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -DMETAGRAPH_BUILD_EXECUTABLE=OFF
+)
+
+# Forward optional flags if they're defined
+if(DEFINED BUILD_STATIC)
+    list(APPEND METAGRAPH_CMAKE_ARGS -DBUILD_STATIC=${BUILD_STATIC})
+endif()
+if(DEFINED WITH_AVX)
+    list(APPEND METAGRAPH_CMAKE_ARGS -DWITH_AVX=${WITH_AVX})
+endif()
+if(DEFINED WITH_MSSE42)
+    list(APPEND METAGRAPH_CMAKE_ARGS -DWITH_MSSE42=${WITH_MSSE42})
+endif()
```

Comment:
Can we make these `true` by default, or enable autodetection?
Otherwise, it might be easy to miss this and lose in the final performance.